docs/source/optim.rst

```diff
@@ -270,16 +270,44 @@ SWA has been proposed in `Averaging Weights Leads to Wider Optima and Better Gen
 Constructing averaged models
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-`AveragedModel` class serves to compute the weights of the SWA model. You can create an
-averaged model by running:
+The `AveragedModel` class serves to compute the weights of the SWA or EMA model.
+
+You can create an SWA averaged model by running:
+
+>>> averaged_model = AveragedModel(model)
 
->>> swa_model = AveragedModel(model)
+EMA models are constructed by specifying the ``multi_avg_fn`` argument as follows:
 
-Here the model ``model`` can be an arbitrary :class:`torch.nn.Module` object. ``swa_model``
+>>> averaged_model = AveragedModel(model, multi_avg_fn=get_ema_multi_avg_fn(decay))
+
+
+Here the model ``model`` can be an arbitrary :class:`torch.nn.Module` object. ``averaged_model``
 will keep track of the running averages of the parameters of the ``model``. To update these
 averages, you can use the :func:`update_parameters` function:
 
->>> swa_model.update_parameters(model)
+>>> averaged_model.update_parameters(model)
+
+Custom averaging strategies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :class:`torch.optim.swa_utils.AveragedModel` computes a running equal average of
+the parameters that you provide, but you can also use custom averaging functions with the
+``avg_fn`` or ``multi_avg_fn`` parameters:
+
+- ``avg_fn`` allows defining a function operating on each parameter tuple (averaged parameter, model parameter) and should return the new averaged parameter.
+- ``multi_avg_fn`` allows defining more efficient operations acting on multiple (averaged parameter, model parameter) tuples at the same time, for example using the ``torch._foreach*`` functions. This function should update the averaged parameters in-place.
```

Comment:
```suggestion
- ``multi_avg_fn`` allows defining more efficient operations acting on a tuple of parameter lists, (averaged parameter list, model parameter list), at the same time, for example using the ``torch._foreach*`` functions. This function must update the averaged parameters in-place.
```
It is more accurate to say the function operates on a tuple of lists instead of lists of tuples, and we should be clear that we expect inplace updates.

Reply:
Fixed